test/src/main/java/org/apache/accumulo/test/BrokenBalancerIT.java

```diff
@@ -0,0 +1,150 @@
+import org.apache.accumulo.core.spi.balancer.TableLoadBalancer;
+import org.apache.accumulo.core.util.UtilWaitThread;
+import org.apache.accumulo.minicluster.ServerType;
+import org.apache.accumulo.miniclusterImpl.MiniAccumuloConfigImpl;
+import org.apache.accumulo.test.functional.ConfigurableMacBase;
+import org.apache.accumulo.test.util.Wait;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.Text;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BrokenBalancerIT extends ConfigurableMacBase {
+
+  private static final Logger log = LoggerFactory.getLogger(BrokenBalancerIT.class);
+
+  public static class BrokenBalancer extends SimpleLoadBalancer {
+    public BrokenBalancer() {
+      super();
+    }
+
+    public BrokenBalancer(TableId tableId) {
+      super(tableId);
+    }
+
+    @Override
+    public void init(BalancerEnvironment balancerEnvironment) {
+      throw new IllegalStateException();
+    }
+  }
+
+  @Override
+  public void configure(MiniAccumuloConfigImpl cfg, Configuration hadoopCoreSite) {
+    Map<String,String> siteConfig = cfg.getSiteConfig();
+    siteConfig.put(Property.TSERV_MAXMEM.getKey(), "10K");
+    siteConfig.put(Property.TSERV_MAJC_DELAY.getKey(), "50ms");
+    siteConfig.put(Property.MANAGER_TABLET_GROUP_WATCHER_INTERVAL.getKey(), "3s");
+    cfg.setSiteConfig(siteConfig);
+    // ensure we have two tservers
+    if (cfg.getNumTservers() != 2) {
+      cfg.setNumTservers(2);
+    }
+  }
+
+  @Test
+  public void testBalancerException() throws Exception {
+    String tableName = getUniqueNames(1)[0];
+    testBadBalancer(BrokenBalancer.class.getName(), tableName);
+  }
+
+  @Test
+  public void testBalancerNotFound() throws Exception {
+    String tableName = getUniqueNames(1)[0];
+    testBadBalancer("org.apache.accumulo.abc.NonExistentBalancer", tableName);
+  }
+
+  private void testBadBalancer(String balancerClass, String tableName) throws Exception {
+    var executor = Executors.newCachedThreadPool();
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProperties()).build()) {
+      SortedSet<Text> splits = new TreeSet<>();
+      for (int i = 0; i < 10; i++) {
+        splits.add(new Text("" + i));
+      }
+      var props = Map.of(Property.TABLE_LOAD_BALANCER.getKey(), balancerClass);
+      NewTableConfiguration ntc =
+          new NewTableConfiguration().withSplits(splits).setProperties(props);
+      c.tableOperations().create(tableName, ntc);
+
+      var scanFuture = executor.submit(() -> {
+        try (var scanner = c.createScanner(tableName)) {
+          scanner.forEach((k, v) -> System.out.println(k + " " + v));
+        }
+        return 0;
+      });
+
+      UtilWaitThread.sleep(5000);
+      // scan should not be able to complete because the tablet should not be assigned
+      Assertions.assertFalse(scanFuture.isDone());
```

Comment:
Curious if you could replace with:
```
      assertEquals(0, BalanceIT.countLocations(c, tableName).size());
```

Reply:
was able to do something similar in fe8f746 and it simplified the test.